source/Waves/WaveGrid.cpp

```diff
@@ -0,0 +1,897 @@
+		std::stringstream ss;
+		ss << "constructSteadyCurrentGrid failed to read currents_profile.txt "
+		      "file: "
+		   << err.what();
+		throw input_file_error(ss.str().c_str());
+	}
+
+	if (lines.size() < 4) {
+		LOGERR << "The file '" << CurrentsFilename
+		       << "' should have at least 4 lines" << endl;
+		throw moordyn::input_file_error("Invalid file format");
+	}
+
+	vector<real> UProfileZ;
+	vector<real> UProfileUx;
+	vector<real> UProfileUy;
+	vector<real> UProfileUz;
+
+	for (unsigned int i = 3; i < lines.size(); i++) {
+		vector<string> entries = moordyn::str::split(lines[i]);
+		if (entries.size() < 2) {
+			LOGERR << "The file '" << CurrentsFilename
+			       << "' should have at least 2 columns" << endl;
+			throw moordyn::input_file_error("Invalid file format");
+		}
+		UProfileZ.push_back(stod(entries[0]));
+		UProfileUx.push_back(stod(entries[1]));
+
+		if (entries.size() >= 3)
+			UProfileUy.push_back(stod(entries[2]));
+		else
+			UProfileUy.push_back(0.0);
+
+		if (entries.size() >= 4)
+			UProfileUz.push_back(stod(entries[3]));
+		else
+			UProfileUz.push_back(0.0);
+	}
+	LOGMSG << "'" << CurrentsFilename << "' parsed" << endl;
+
+	// NOTE: check data
+
+	std::vector<real> px = { 0.0 };
+	std::vector<real> py = { 0.0 };
+	// set 1 time step to indicate steady data
+	unsigned int nt = 1;
+	real dtWave = 1.0; // arbitrary entry
+
+	auto currentGrid =
+	    make_unique<CurrentGrid>(_log, px, py, UProfileZ, nt, dtWave);
+	currentGrid->allocateKinematicArrays();
+
+	// fill in output arrays
+	for (unsigned int i = 0; i < currentGrid->nz; i++) {
+		currentGrid->CurrentVel()[0][0][i][0] =
+		    vec3(UProfileUx[i], UProfileUy[i], UProfileUz[i]);
+	}
+	return std::move(currentGrid);
+}
+
+std::unique_ptr<CurrentGrid>
+constructDynamicCurrentGrid(const std::string& folder,
+                            const EnvCondRef env,
+                            moordyn::Log* _log)
+{
+
+	const string CurrentsFilename = folder + "/current_profile_dynamic.txt";
+	LOGMSG << "Reading currents dynamic profile from '" << CurrentsFilename
+	       << "'..." << endl;
+
+	vector<string> lines;
+
+	try {
+		lines = moordyn::fileIO::fileToLines(CurrentsFilename);
+	} catch (moordyn::input_file_error& err) {
+		LOGERR << "Cannot read the file '" << CurrentsFilename << "'" << endl;
+		std::stringstream ss;
+		ss << "Waves::setup failed to read currents_profile_dynamic.txt "
+		      "file: "
+		   << err.what();
+		throw input_file_error(ss.str().c_str());
+	}
+
+	if (lines.size() < 7) {
+		LOGERR << "The file '" << CurrentsFilename
+		       << "' should have at least 7 lines" << endl;
+		throw moordyn::input_file_error("Invalid file format");
+	}
+
+	vector<real> UProfileZ;
+	vector<real> UProfileT;
+	vector<vector<real>> UProfileUx;
+	vector<vector<real>> UProfileUy;
+	vector<vector<real>> UProfileUz;
+
+	// this is the depths row
+	vector<string> entries = moordyn::str::split(lines[4]);
```

Comment:
[Now a test failed](https://github.com/FloatingArrayDesign/MoorDyn/actions/runs/5014577378/jobs/9004047835?pr=71). The problem are the whitespaces at the tail of tests/Mooring/wavekin_3/current_profile_dynamic.txt. Probably we should add and use [a strip method](https://stackoverflow.com/questions/216823/how-to-trim-an-stdstring)